src/Json/Value.php

```diff
@@ -647,25 +648,67 @@ public function ¿timestamp(): ?DateTimeImmutable
      *                                   $className.
      * @throws Exception\Runtime         If $className does not exist, or does not extend Enum.
      */
-    public function enum(string $className): Enum
+    public function enum(string $className): Enum | BackedEnum
```

Comment:
Je moet denk ik ook een check kunnen toevoegen voor native enums dat als het wél een `UnitEnum` is, en niet een `StringBackedEnum`, dat hij dan een exception gooit dat alleen string backed enums ondersteund worden.

Reply:
Waarom niet ook IntBackedEnum's ondersteunen 🙃

Reply:
:facepalm: inderdaad.

Reply:
problem solved: the additional $this->string() is unneeded because BackedEnum::from() already throws ValueError if the data-type is incorrect (strict type comparison with the backed values)